### PR TITLE
Simplified the RibbonTabItem template for the office 2010 theme.

### PR DESCRIPTION
**Pull request by abdelkarim**
This Pull Request is the result of what we discussed [here](https://github.com/fluentribbon/Fluent.Ribbon/issues/84), please review and let me know of any requires changes.

The goals of this Pull Request are:
* An easy to understand and customize template.
* A better performance, by reducing the visual tree complexity.

The goals are achieved by using:
* The **RibbonTabItemBorder** element will take care of creating and rendering the geometries of the selected and mouse over states, instead of combining a **Grid** and various elements to the achieve the same result.
* The **SingleCellGrid** panel is used to group related elements, instead of the **Grid** panel.

### CODE REVIEW

**Comment by batzen** [comment]
What's your plan for the Office 2013 and Windows 8 themes?
I thought you would support all themes and not just one.

**Comment by abdelkarim** [comment]
The `RibbonTabItemBorder` will not be used in the **Office 2013** and **Windows 8** themes, because those themes have flat corners and no nested borders. --> a simple `Border` will do the job.

I will review those templates and send you another Pull Request.

**Comment by abdelkarim** [comment]
Can you please review and merge this so that I can send you another.
Some feedback will be welcomed :)

**Comment by batzen** [comment]
Answered in #84

File: Fluent/Controls/Primitives/RibbonTabItemBorder.cs [hunk 1]
```diff
@@ -0,0 +1,771 @@
...
+        /// Identifies the <see cref="CornerRadius"/> dependency property.
+        /// </summary>
+        public static readonly DependencyProperty CornerRadiusProperty = DependencyProperty.Register(
+            "CornerRadius",
+            typeof(double),
+            typeof(RibbonTabItemBorder),
+            new FrameworkPropertyMetadata(
+                0.0,
+                FrameworkPropertyMetadataOptions.AffectsMeasure | FrameworkPropertyMetadataOptions.AffectsRender,
+                (o, args) =>
```
Comment (batzen) [inline Fluent/Controls/Primitives/RibbonTabItemBorder.cs]: Please don't use inline functions in DP-definitions.

File: Fluent/Controls/Primitives/RibbonTabItemBorder.cs [hunk 2]
```diff
@@ -0,0 +1,771 @@
...
+        /// Identifies the <see cref="BorderBrush"/> dependency property.
+        /// </summary>
+        public static readonly DependencyProperty BorderBrushProperty = DependencyProperty.Register(
+            "BorderBrush",
+            typeof(Brush),
+            typeof(RibbonTabItemBorder),
+            new FrameworkPropertyMetadata(
+                null,
+                FrameworkPropertyMetadataOptions.AffectsRender,
+                (o, args) =>
```
Comment (batzen) [inline Fluent/Controls/Primitives/RibbonTabItemBorder.cs]: Please don't use inline functions in DP-definitions.

File: Fluent/Controls/Primitives/RibbonTabItemBorder.cs [hunk 3]
```diff
@@ -0,0 +1,771 @@
...
+        /// Identifies the <see cref="InnerBorderBrush"/> dependency property.
+        /// </summary>
+        public static readonly DependencyProperty InnerBorderBrushProperty = DependencyProperty.Register(
+            "InnerBorderBrush",
+            typeof(Brush),
+            typeof(RibbonTabItemBorder),
+            new FrameworkPropertyMetadata(
+                null,
+                FrameworkPropertyMetadataOptions.AffectsRender,
+                (o, args) =>
```
Comment (batzen) [inline Fluent/Controls/Primitives/RibbonTabItemBorder.cs]: Please don't use inline functions in DP-definitions.

File: Fluent/Controls/Primitives/RibbonTabItemBorder.cs [hunk 4]
```diff
@@ -0,0 +1,771 @@
...
+        /// Identifies the <see cref="BorderThickness"/> dependency property.
+        /// </summary>
+        public static readonly DependencyProperty BorderThicknessProperty = DependencyProperty.Register(
+            "BorderThickness",
+            typeof(double),
+            typeof(RibbonTabItemBorder),
+            new FrameworkPropertyMetadata(
+                0.0,
+                FrameworkPropertyMetadataOptions.AffectsMeasure | FrameworkPropertyMetadataOptions.AffectsRender,
+                (o, args) =>
```
Comment (batzen) [inline Fluent/Controls/Primitives/RibbonTabItemBorder.cs]: Please don't use inline functions in DP-definitions.

**Comment by batzen** [comment]
Please just take those comments and don't fix anything for now as i am still unsure if i should accept this PR, so investing more time might be wasted.

**Comment by abdelkarim** [comment]
I am not doing anything until you give a final reply.

**Comment by batzen** [comment]
I finally had some time to give your changes a try.
I decided to reject the changes because of the following:
- It may be faster, but it introduces additional complexity. Maintaining or editing a few borders in XAML seems much easier/flexible than having to change the control.
- If someone wants to modify/create a theme it's quite easy now and only involves changing XAML, with the new Control one would have to clone the class and implement the changes one needs.
- Your solution introduces a few visual issues and I, personally, don't want to risk the stability of the Office 2010 theme.

I have to admit that I should have asked more detailed questions at the time you asked for my opinion and I am sorry that your time was "wasted". If you ever happen to visit Germany I would be happy to buy you some beer to compensate my mistake.

**Comment by abdelkarim** [comment]
I have to say that I am surprised but I expected this answer somehow when you hesitated after I submitted my PR.

Complexity ? I wonder if the WPF team made it more complex when they wrote **ButtonChrome** and **BulletChrome** ....etc to simplify and speed the **Button**, **CheckBox** and **RadioButton**. My Pull Request is inspired by this.

Why would they want to clone the class ??

The visual issue has been clearly identified, and it is not introduced by me. The way the header of the **RibbonTabItem**, is being centered, is what introduced the bug in the first place. --> My work made it clearly visible. The bug was fixed in my other branch where I did a re-write of the **TabsContainer**.

Regards,

**Pull request closed without merging**